tas/views.py

```diff
@@ -43,73 +42,31 @@ def get_departments_json(request):
-    if is_federated:
-        problem_description = "This PI Eligibility request can be reviewed at " +\
-            "https://wwww.chameleoncloud.org/admin/chameleon/pieligibility/"
-    else:
-        problem_description = ""
-    ticket = rtModels.Ticket(subject = subject,
-                             problem_description = problem_description,
-                             requestor = "us@tacc.utexas.edu")
-    rt.createTicket(ticket)
-
 @login_required
 def profile_edit(request):
-    is_federated = request.session.get('is_federated', False)
-    kwargs = {'is_federated':is_federated}
-    if is_federated:
-        user_info = {}
-        user_info['firstName'] = request.user.first_name
-        user_info['lastName'] = request.user.last_name
-        user_info['email'] = request.user.email
-    else:
-        tas = TASClient()
-        user_info = tas.get_user(username=request.user)
-    if is_federated:
-        piEligibility = request.user.pi_eligibility()
-    else:
-        piEligibility = user_info['piEligibility']
+    mapper = ProjectAllocationMapper(request)
+    user_info = mapper.get_user(request.user)
 
     if request.method == 'POST':
         request_pi_eligibility = request.POST.get('request_pi_eligibility')
         form = UserProfileForm(request.POST, initial=user_info)
-        '''
-            if user is federated only check/update PI Eligibility Requests
-        '''
-        if is_federated and request_pi_eligibility:
-            pie_request = PIEligibility()
-            pie_request.requestor = request.user
-            pie_request.save()
-            _create_ticket_for_pi_request(user_info, is_federated)
-            messages.success(request, 'Your profile has been updated!')
-            return HttpResponseRedirect(reverse('tas:profile'))
-        '''
-            If not federated validate form and update TAS profile info
-        '''
+        if request_pi_eligibility:
+            form.pi_eligible()
+
         if form.is_valid():
             data = form.cleaned_data
-            if request_pi_eligibility:
-                    data['piEligibility'] = 'Requested'
-                    _create_ticket_for_pi_request(user_info)
-            else:
-                data['piEligibility'] = user_info['piEligibility']
-            data['source'] = 'Chameleon'
-            tas.save_user(user_info['id'], data)
+            mapper.update_user_profile(user_info, data, request_pi_eligibility)
             messages.success(request, 'Your profile has been updated!')
             return HttpResponseRedirect(reverse('tas:profile'))
     else:
-        form = UserProfileForm(initial=user_info,**kwargs)
+        form = UserProfileForm(initial=user_info)
+        if user_info['piEligibility'].upper() == 'ELIGIBLE':
+            form.pi_eligible()
 
     context = {
         'form': form,
         'user': user_info,
-        'piEligibility': piEligibility,
+        'piEligibility': user_info['piEligibility'],
```

Comment:
Watch out for the letter case here... I hit a bug with this at some point, where the template was expecting it to look like `"ELIGIBLE"` but it was actually `"Eligible"`.